src/libraries/Pool.sol

```diff
@@ -444,16 +444,17 @@ library Pool {
             self.feeGrowthGlobal1X128 = state.feeGrowthGlobalX128;
         }
 
+        // in an unchecked block so sign flipping is not safe without a helper function
         unchecked {
             if (params.zeroForOne == exactInput) {
                 result = toBalanceDelta(
-                    (params.amountSpecified - state.amountSpecifiedRemaining).toInt128(),
-                    state.amountCalculated.toInt128()
+                    (params.amountSpecified - state.amountSpecifiedRemaining).toInt128().flipSign(),
```

Comment:
Hm.. Since we are flipping amountSpecified this will not be needed correct?

Reply:
Like imo all the math should be changed to return the correct sign and something like flipSign shoudn't be needed

Reply:
Right yeah it gets fixed in emily's PR if you can review that too! 🙌 Then we can merge that into this PR